salvo/tools/coverage.sh

```diff
@@ -35,6 +35,11 @@ CMD="${CMD} -o coverage/coverage.dat"
 
 # Extract all coverage data for salvo project files
 lcov -e coverage/coverage.dat -o coverage/salvo.dat '*/salvo/src/lib/*.py'
+if [ "$?" != "0" ]; then
```

Comment:
Checking the error after individual commands may still be error prone (we may forget to add checks when we add new commands).

Can we do it at the top of the file instead? This file could start as:

```
#!/bin/bash

# Exit immediately if a command exits with a non-zero status.
# pipefail indicates that the return value of a pipeline is the status
# of the last command to exit with a non-zero status.
set -eo pipefail
```

We may want to adopt this approach in all our CI scripts.

Reply:
@mum4k Thanks for pointing out this better approach.